ravem/tests/operations/connect_test.py

```diff
@@ -0,0 +1,412 @@
+                        "name": "videoconference",
+                    }
+                ],
+            }
+        ),
+    )
+    httpretty.register_uri(
+        httpretty.POST,
+        RAVEM_TEST_API_ENDPOINT + service_type + "/disconnect",
+        status=200,
+        content_type="application/json",
+        body=json.dumps({"result": "OK"}),
+    )
+
+    vc_room = MagicMock()
+    vc_room.type = service_type
+    vc_room.data = data
+
+    with pytest.raises(RavemException) as excinfo:
+        connect_room(room_name, vc_room, force=True)
+
+    assert (
+        str(excinfo.value)
+        == "Failed to disconnect the room {0} from the videoconference room {1} "
+           "with a unknown error".format(room_name, different_vc_room)
+    )
+    log = extract_logs(caplog, one=True, name="indico.plugin.ravem")
+    assert (
+        log.message
+        == "Failed to disconnect the room {0} from the videoconference room {1} "
+           "with a unknown error".format(room_name, different_vc_room)
+    )
+
+    assert len(httpretty.httpretty.latest_requests) == 2 + RavemPlugin.settings.get(
+        "polling_limit"
+    )
+    request = httpretty.httpretty.latest_requests[1]
+    assert request.path.startswith(RAVEM_TEST_PATH + service_type + "/disconnect")
+    assert request.parsed_body == {"roomName": room_name}
+
+
+@pytest.mark.usefixtures("db")
+@pytest.mark.parametrize(*gen_params(connected_fixtures, "room_name", "service_type", "connected", "data"))
+def test_connect_room_force_error(caplog, httpretty, room_name, service_type, connected, data):
+    RavemPlugin.settings.set("api_endpoint", RAVEM_TEST_API_ENDPOINT)
+    error_message = "Some internal error"
+    different_vc_room = "different_vc_room"
+    httpretty.register_uri(
+        httpretty.GET,
+        RAVEM_TEST_API_ENDPOINT + "rooms/details",
+        status=200,
+        content_type="application/json",
+        body=json.dumps(
+            {
+                "roomName": room_name,
+                "deviceType": service_type,
+                "services": [
+                    {
+                        "status": connected,
+                        "eventName": different_vc_room,
+                        "name": "videoconference",
+                    }
+                ],
+            }
+        ),
+    )
+    httpretty.register_uri(
+        httpretty.POST,
+        RAVEM_TEST_API_ENDPOINT + service_type + "/disconnect",
+        status=200,
+        content_type="application/json",
+        body=json.dumps({"error": error_message}),
+    )
+
+    vc_room = MagicMock()
+    vc_room.type = service_type
+    vc_room.data = data
+
+    with pytest.raises(RavemException) as excinfo:
+        connect_room(room_name, vc_room, force=True)
+
+    assert (
+        str(excinfo.value)
+        == "Failed to disconnect the room {0} from the videoconference room {1} with error: {2}".format(
+        room_name, different_vc_room, error_message
+    )
+    )
```

Comment:
please tell me black is not really generating such garbage...

Reply:
I don't think it was black.